client/src/components/Routes/PrivateRoute.js

```diff
@@ -1,25 +1,10 @@
 import React, { useEffect, useState } from "react";
 import { Route, Redirect } from "react-router-dom";
-import { testAuth } from "../../utils/googleAuth";
+import { useAuth } from "../../providers/AuthProvider";
 
 const PrivateRoute = ({ children, ...rest }) => {
-  const [authenticated, setAuthenticated] = useState(false);
-  const [redirect, setRedirect] = useState(false);
-
-  useEffect(() => {
-    testAuth()
-      .then((res) => {
-        setAuthenticated(res);
-      })
-      .catch(() => {
-        setRedirect(true);
-      });
-  }, []);
-  if (redirect) {
-    return <Redirect to="/login" />;
-  } else {
-    return authenticated && <Route {...rest}>{children}</Route>;
-  }
+  const authenticated = useAuth();
+  return authenticated && <Route {...rest}>{children}</Route>;
```

Comment:
Why was the Redirect removed? We would still need that right?

Reply:
This was more like a last minute fix. The issue is that when user lands on protected route on a page refresh, they would get redirected before the context even had a chance to grab from server whether they are authenticated. 

This is definitely something that needs to be improved, though.

Reply:
Okay, I'll make a new issue for it